src/EditorFeatures/TestUtilities/Diagnostics/AbstractUserDiagnosticTest.cs

```diff
@@ -199,27 +199,9 @@ protected Document GetDocumentAndAnnotatedSpan(TestWorkspace workspace, out stri
             }
 
             var diagnostic = diagnostics.First();
-            Func<Document, ImmutableHashSet<string>, CancellationToken, Task<IEnumerable<Diagnostic>>> getDocumentDiagnosticsAsync =
-                async (d, diagIds, c) =>
-                {
-                    var root = await d.GetSyntaxRootAsync();
-                    var diags = await testDriver.GetDocumentDiagnosticsAsync(provider, d, root.FullSpan);
-                    diags = diags.Where(diag => diagIds.Contains(diag.Id));
-                    return diags;
-                };
-
-            Func<Project, bool, ImmutableHashSet<string>, CancellationToken, Task<IEnumerable<Diagnostic>>> getProjectDiagnosticsAsync =
-                async (p, includeAllDocumentDiagnostics, diagIds, c) =>
-                {
-                    var diags = includeAllDocumentDiagnostics
-                        ? await testDriver.GetAllDiagnosticsAsync(provider, p)
-                        : await testDriver.GetProjectDiagnosticsAsync(provider, p);
-                    diags = diags.Where(diag => diagIds.Contains(diag.Id));
-                    return diags;
-                };
-
             var diagnosticIds = ImmutableHashSet.Create(diagnostic.Id);
-            var fixAllDiagnosticProvider = new FixAllState.FixAllDiagnosticProvider(diagnosticIds, getDocumentDiagnosticsAsync, getProjectDiagnosticsAsync);
+            var fixAllDiagnosticProvider = new FixAllDiagnosticProvider(provider, testDriver, diagnosticIds);
```

Comment:
Note: this approach in tests seems wonky to me.  Why are we not just going through the codepath that the actual CodeFixService goes through?  In other words, it seems to me like the test code is reeimplementing a lot of logic that we already have in product code. We should just reuse that logic, instead of doing things our own way here.

This sort of testing approach gives me less confidence that we're actually validating the code that will be running in the product.